tests/integration/schema_registry/test_proto_serializers.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+
+from confluent_kafka import TopicPartition
+from confluent_kafka.error import ConsumeError
+from confluent_kafka.schema_registry.protobuf import ProtobufSerializer, ProtobufDeserializer
+from ..schema_registry.gen import NestedTestProto_pb2, TestProto_pb2, \
+    PublicTestProto_pb2
+from tests.integration.schema_registry.gen.DependencyTestProto_pb2 import DependencyMessage
+from tests.integration.schema_registry.gen.exampleProtoCriteo_pb2 import ClickCas
+
+
+@pytest.mark.parametrize("pb2, data", [
+    (TestProto_pb2.TestMessage, {'test_string': "abc",
+                                 'test_bool': True,
+                                 'test_bytes': b'look at these bytes',
+                                 'test_double': 1.0,
+                                 'test_float': 12.0}),
+    (PublicTestProto_pb2.TestMessage, {'test_string': "abc",
+                                       'test_bool': True,
+                                       'test_bytes': b'look at these bytes',
+                                       'test_double': 1.0,
+                                       'test_float': 12.0}),
+    (NestedTestProto_pb2.NestedMessage, {'user_id':
+                                         NestedTestProto_pb2.UserId(
+                                             kafka_user_id='oneof_str'),
+                                         'is_active': True,
+                                         'experiments_active': ['x', 'y', '1'],
+                                         'status': NestedTestProto_pb2.INACTIVE,
+                                         'complex_type':
+                                             NestedTestProto_pb2.ComplexType(
+                                                 one_id='oneof_str',
+                                                 is_active=False)})
+])
+def test_protobuf_message_serialization(kafka_cluster, pb2, data):
+    """
+    Validates that we get the same message back that we put in.
+
+    """
+    topic = kafka_cluster.create_topic("serialization-proto")
+    sr = kafka_cluster.schema_registry({'url': 'http://localhost:8081'})
```

Comment:
I'd rather see us use a dummy-url that does not risk being an existing SR instance, e.g., http://localhost:65534.
Otherwise we risk messing up someones unrelated dev SR.

Reply:
yes sorry this is just there for running integration tests right now since there is no SR image with 5.5. I will remove it next push after running tests.

Reply:
In that case maybe we need to make it configurable to use a non-trivup SR instead of hard-coding the URL in all these tests?   
This will be useful any time we need to test with a non-released version of SR, so how about adding this to trivup, maybe have the trivup SchemaRegistry honour env `TRIVUP_EXISTING_SR_URL` to expose that URL through its .url property rather than starting an SR?

Reply:
Yes I think it would be great if trivup would accept overrides from environment variables prefixed with `TRIVUP_` (to avoid name clashing). I'd rather not bake this into the Python client tests as its applicable to anything using Trivup. I'd like  SR  to be  treated as a first class cluster as well.  I'll submit a PR after all of this is done and merged